### ISSUE TITLE
Routing Resolver/Compiler rejects URL that contain filenames with extension

### ISSUE THREAD

**Issue by DiegoPino** [trellis-ldp/static-ldp#25]
Symfony internals define some built in separators for arguments when building their routing resolution. Allowing `/` to exists is already solved, but dots == `.` used for filename extensions break the URL because the compiler sees whats comes after, a.k.a the file extension as a different argument.

I did some research and found the "where":

https://github.com/symfony/routing/blob/3.2/RouteCompiler.php#L101

Seems like that i also know the "how" now.

I would suggest we also move the base route resolutor (like adding a /ldp/) one level up to allow a different strategy for files outside of the LDP tree, like if you want to serve a `JSON-LD @context` as a simple file?

**Comment by acoburn** [comment]
@DiegoPino regarding moving the base route resoluter, wouldn't it be possible to just install the application in a subdirectory? Alternately, it would be possible to `mount` the routes at a configurable location (e.g. `/ldp` or `/`)

**Comment by DiegoPino** [comment]
@acoburn totally the mount option. Its nothing else than that at the silex level, and allowing a different route resolution (if needed) for the parent mount point. If that different, parent routing won't be managed by silex (apache static, etc), then yeah, could be managed on apache and that should do the trick. But all that would be better solved via a config entry in the yaml file and then added to the route mount file to give as much flexibility as possible and allow also apache-less work.

**Comment by whikloj** [comment]
So does this mean that you can't browse to a URL like 
`http://static-ldp.localhost/2012_12_02/moreFiles/object.ttl` 
or 
`http://static-ldp.localhost/2012_12_02/moreFiles/IMG_0084.JPG`?

**Comment by acoburn** [comment]
I don't have any problem browsing to nested files.

**Comment by whikloj** [comment]
Also to go along with the above question are you using a full webserver or PHP's internal one, because if it's the latter I found this ticket
https://github.com/silexphp/Silex/issues/1103

Which seems related.

**Comment by DiegoPino** [comment]
@whikloj not sure if i'm getting those recommendations or i'm not looking at the correct spot or my english makes no sense: we don't want to serve static files, we want to handle an URL that contains a static file like

 `http://static-ldp.localhost/2012_12_02/moreFiles/IMG_0084.JPG` 

as an PHP URL we can delegate to our Controller . 

Not sure, maybe it's my environment but if i have here 
`2012_12_02/moreFile/jared/morejared/image.tiff` and that one does not resolve. I get an 4xx and managed by silex (html) even if i have the structure. 
If i rename image.tiff to image and try `2012_12_02/moreFile/jared/morejared/image` then all works.

**Comment by DiegoPino** [comment]
@acoburn @whikloj feel free to close this if you don't have this issue.

**Comment by acoburn** [comment]
@DiegoPino you'll want to verify your environment config -- in my tests, I'm using Apache httpd and [this .htaccess file](https://github.com/trellis-ldp/static-ldp/blob/master/.htaccess).

**Comment by DiegoPino** [comment]
@acoburn ok, will close then. I never use Apache while developing in silex, because i just have too many projects at the same time. Thanks!

**Comment by whikloj** [comment]
@DiegoPino I would say that the issue described is exactly what you are experiencing. 

We are _serving_ static files,  in that the Silex framework we have built (which is itself dynamic) is serving a directory of static files. 

However I just tried using the PHP internal server and this and it seemed to work for me. 
I did
> [/sw/var/www/trellis-ldp] 
\> php -S localhost:8888 -t static-ldp static-ldp/index.php
PHP 7.1.3 Development Server started at Wed Apr  5 11:21:38 2017
Listening on http://localhost:8888
Document root is /sw/var/www/trellis-ldp/static-ldp
Press Ctrl-C to quit.

then I was able to browse to `http://localhost:8888/2012_12_02/moreFiles/object.ttl` or `http://localhost:8888/2012_12_02/moreFiles/IMG_0082.JPG`
